File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x >= 0)
        {
        vector<int> digits;
        while(x)
        {
            int digit = x % 10;
            digits.push_back(digit);
            x = x / 10;
        }
        int size = digits.size();
        int i=0;
        int j = size ? size - 1 : 0;
        bool isPal = true;
        if (i >= 0 && j >=0)
        {
            while (i <= j)
            {
                if ( i == 0 && j == 0)
                {
                    break;                      
                }
                if (digits[i] == digits[j])
                {
                    i++;
                    j--;
                    continue;                
                }
                else
                {
                    isPal = false;
                    break;                
                }
            }
        }
        else
        {
            isPal = false;            
        }
        return isPal;
        }
        else
        {
            return false;
        }
    }
};

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String num = Integer.toString(x);
        
        int l = num.length();
        int i = 0;
        int j = l-1;
        
        while (i < j) {
            if (num.charAt(i) == num.charAt(j)) {
                i++;
                j--;
                continue;
            } else {
                return false;
            }
        }
        return true;
    }
}